File: internal/models/game.go
package models

type PlayTime struct {
	Hastily    int `json:"hastily"`
	Normally   int `json:"normally"`
	Completely int `json:"completely"`
}

type Game struct {
	ID               string     `json:"id"`
	Title            string     `json:"title"`
	ContentType      string     `json:"type"`
	Description      *string    `json:"description,omitempty"`
	ImageURL         *string    `json:"image_url,omitempty"`
	GameType         *string    `json:"game_type,omitempty"`
	ReleaseDate      *string    `json:"release_date,omitempty"`
	Authors          []Author   `json:"authors,omitempty"`
	Platforms        []Platform `json:"platforms,omitempty"`
	Genres           []string   `json:"genres,omitempty"`
	Themes           []string   `json:"themes,omitempty"`
	GameModes        []string   `json:"game_modes,omitempty"`
	Series           *string    `json:"series,omitempty"`
	AgeRating        *string    `json:"age_rating,omitempty"`
	PlayTime         *PlayTime  `json:"play_time,omitempty"`
	Images           *Images    `json:"-"`
}

type GameResponse struct {
	ID               string       `json:"id"`
	Type             string       `json:"type"`
	Title            string       `json:"title"`
	GameType         *string      `json:"game_type,omitempty"`
	Description      *string      `json:"description,omitempty"`
	ImageURL         *string      `json:"image_url,omitempty"`
	ReleaseDate      *string      `json:"release_date,omitempty"`
	Authors          []Author     `json:"authors,omitempty"`
	Platforms        []Platform   `json:"platforms,omitempty"`
	Genres           []string     `json:"genres,omitempty"`
	Themes           []string     `json:"themes,omitempty"`
	GameModes        []string     `json:"game_modes,omitempty"`
	Series           *string      `json:"series,omitempty"`
	AgeRating        *string      `json:"age_rating,omitempty"`
	PlayTime         *PlayTime    `json:"play_time,omitempty"`
	Images           []ImageEntry `json:"images,omitempty"`
}

func (g *Game) ToResponse(imagesSize int) GameResponse {
	resp := GameResponse{
		ID:          g.ID,
		Type:        g.ContentType,
		Title:       g.Title,
		GameType:    g.GameType,
		Description: g.Description,
		ImageURL:    g.ImageURL,
		ReleaseDate: g.ReleaseDate,
		Authors:     g.Authors,
		Platforms:   g.Platforms,
		Genres:      g.Genres,
		Themes:      g.Themes,
		GameModes:   g.GameModes,
		Series:           g.Series,
		AgeRating:        g.AgeRating,
		PlayTime:         g.PlayTime,
	}

	if g.Images != nil {
		resp.Images = g.Images.ToList(imagesSize)
	}

	return resp
}

File: content/services/local_content_store/mappers/game.py
"""Map a normalized game payload onto GameDetail + GamePlatform + taxonomies."""
from __future__ import annotations

from typing import Any, Dict, List, Optional

from django.db import transaction

from content.models import (
    ContentItem,
    GameDetail,
    GamePlatform,
)
from content.services.payload_helpers import hash_payload, parse_iso_date

from ._common import (
    replace_content_item_authors,
    replace_images,
    require_payload_shape,
    sync_game_taxonomies,
)


_MAPPED_KEYS = (
    'id', 'type', 'title', 'game_type', 'description', 'image_url',
    'release_date', 'series', 'authors', 'platforms', 'genres',
    'themes', 'game_modes', 'play_time', 'images',
)


def upsert(
    content_item: ContentItem,
    payload: Dict[str, Any],
    *,
    request_country: Optional[str] = None,
) -> None:
    require_payload_shape(payload, expected_type='game')

    subset = {k: payload.get(k) for k in _MAPPED_KEYS if k in payload}
    payload_hash = hash_payload(subset)

    play_time = payload.get('play_time') or {}
    play_min = play_time.get('min') if isinstance(play_time, dict) else None
    play_max = play_time.get('max') if isinstance(play_time, dict) else None

    defaults = {
        'title': payload.get('title') or '',
        'game_type': payload.get('game_type') or '',
        'description': payload.get('description') or '',
        'image_url': payload.get('image_url') or '',
        'release_date': parse_iso_date(payload.get('release_date')),
        'series': payload.get('series') or '',
        'play_time_min': play_min,
        'play_time_max': play_max,
        'source_payload_hash': payload_hash,
    }

    raw_platforms = payload.get('platforms')

    with transaction.atomic():
        game_detail, _ = GameDetail.objects.update_or_create(
            content_item=content_item,
            defaults=defaults,
        )
        replace_content_item_authors(content_item, payload)
        replace_images(content_item, payload)
        sync_game_taxonomies(game_detail, payload)

        GamePlatform.objects.filter(game_detail=game_detail).delete()
        rows: List[GamePlatform] = []
        if isinstance(raw_platforms, list):
            seen = set()
            for entry in raw_platforms:
                if not isinstance(entry, dict):
                    continue
                name = entry.get('name')
                if not name or name in seen:
                    continue
                seen.add(name)
                rows.append(
                    GamePlatform(
                        game_detail=game_detail,
                        name=name,
                        image_url=entry.get('image_url') or '',
                    )
                )
        if rows:
            GamePlatform.objects.bulk_create(rows)
